src/main/java/com/uber/cadence/internal/replay/WorkflowContext.java

```diff
@@ -131,4 +136,14 @@ void setCurrentRunId(String currentRunId) {
   String getCurrentRunId() {
     return currentRunId;
   }
+
+  long getScheduleTimeMillis() {
+    return TimeUnit.NANOSECONDS.toMillis(
+        decisionTask.getHistory().getEvents().get(0).getTimestamp());
+  }
+
+  long getExecutionTimeMillis() {
+    return TimeUnit.NANOSECONDS.toMillis(
+        decisionTask.getHistory().getEvents().get(1).getTimestamp());
```

Comment:
I don't think it is always true. For example signalWithStart will have signal as a second event. 
Also when cron workflow is waiting for exectution all received signals will be put before DecisionTaskScheduled.